src/normalizer/chinese.rs

```diff
@@ -1,26 +1,51 @@
 use std::borrow::Cow;
 
-use character_converter::traditional_to_simplified;
+use pinyin::ToPinyin;
 
 use super::{Normalizer, NormalizerOption};
 use crate::detection::{Language, Script};
 use crate::Token;
 
-/// Normalize Chinese characters by converting them into Simplified Chinese characters.
+/// Normalize Chinese characters by converting them into Pinyin characters.
 ///
-/// This Normalizer uses [`character_converter`] internally to normalize the provided token.
+/// This Normalizer uses [`pinyin`] internally to normalize the provided token.
 pub struct ChineseNormalizer;
 
 impl Normalizer for ChineseNormalizer {
     fn normalize<'o>(
         &self,
         mut token: Token<'o>,
-        _options: NormalizerOption,
+        options: NormalizerOption,
     ) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
-        if let Cow::Owned(s) = traditional_to_simplified(token.lemma()) {
-            token.lemma = Cow::Owned(s);
+        let mut lemma = String::new();
+        let mut char_map = if options.create_char_map { Some(Vec::new()) } else { None };
+
+        for c in token.lemma().chars() {
+            match c.to_pinyin() {
+                Some(converted) => {
+                    let with_tone = converted.with_tone();
+
+                    char_map
+                        .as_mut()
+                        .map(|char_map| char_map.push((c.len_utf8() as u8, with_tone.len() as u8)));
+
+                    lemma.push_str(with_tone);
+                }
+                None => {
+                    char_map
+                        .as_mut()
+                        .map(|char_map| char_map.push((c.len_utf8() as u8, c.len_utf8() as u8)));
+
+                    lemma.push(c);
+                }
+            }
         }
 
+        token.lemma = Cow::Owned(lemma);
+        token.char_end = token.lemma.chars().count();
+        token.byte_end = token.lemma.len();
```

Comment:
These values are kept in the token in the case the token is normalized, if we want to know the normalized lengths, we can dynamically compute it at any time.
```suggestion
```